File: tenminwalk/spec/app_spec.rb
require 'app'
describe App do
  it "returns true when given ['w', 's', 'e', 'e', 'n', 'n', 'e', 's', 'w', 'w']" do
  app = App.new
  route = ['w', 's', 'e', 'e', 'n', 'n', 'e', 's', 'w', 'w']
  expect(app.tenminuteswalk(route)).to be true
  end

  it "returns true when given ['w', 's', 'e', 'n', 'n', 'e', 's', 'w', 'w', 'w']" do
  app = App.new
  route = ['w', 's', 'e', 'n', 'n', 'e', 's', 'w', 'w', 'w']
  expect(app.tenminuteswalk(route)).to be false
  end

  it "returns true when given ['w', 's', 'e', 's', 's', 'e', 's', 'w', 'n', 'n']" do
  app = App.new
  route = ['w', 's', 'e', 's', 's', 'e', 's', 'w', 'n', 'n']
  expect(app.tenminuteswalk(route)).to be false
  end

  it "returns true when given ['w', 's']" do
  app = App.new
  route = ['w', 's']
  expect(app.tenminuteswalk(route)).to be false
  end


end

File: tenminutewalk/Spec/appSpec.js
describe('App', function() {
  var app;

  beforeEach(function() {
    app = new App();
  });

  describe('App', function() {
    it('returns true if its valid route', function() {
      expect(app.isValidWalk(['w', 's', 'e', 'e', 'n', 'n', 'e', 's', 'w', 'w'])).toBe(true);
    });

  describe('App', function() {
    it('returns false if its invalid route', function() {
        expect(app.isValidWalk(['w', 's', 'e', 'n', 'n', 'e', 's', 'w', 'w', 'w'])).toBe(false);
      });

  describe('App', function() {
    it('returns false if its under 10 minute route', function() {
        expect(app.isValidWalk(['w', 's'])).toBe(false);
    });

  describe('App', function() {
    it('returns false if the route is not at the starting point', function() {
      expect(app.isValidWalk(['w', 's', 'e', 's', 's', 'e', 's', 'w', 'n', 'n'])).toBe(false);
     });
    })
   })
  })
 })
});
